### PR TITLE
CLOUDP-240596: fix search index create watcher for Atlas deployments

### PR DESCRIPTION
**Pull request by tibulca**
## Proposed changes
fix search index create watcher for Atlas deployments

_Jira ticket:_ CLOUDP-240596

Testing:
* Atlas deployment: https://spruce.mongodb.com/task/mongodb_atlas_cli_master_e2e_atlas_deployments_atlas_deployments_atlas_clusters_e2e_patch_07319bbad7ef2b9d622be191947b63f4efc256fd_660c07c0bd7dd000079b2ccc_24_04_02_13_27_29/logs?execution=0
* Local deployment: https://spruce.mongodb.com/task/mongodb_atlas_cli_master_e2e_local_deployments_atlas_deployments_local_noauth_e2e_patch_07319bbad7ef2b9d622be191947b63f4efc256fd_660c07c0bd7dd000079b2ccc_24_04_02_13_27_29/logs?execution=0

### CODE REVIEW

**Review by blva — APPROVED** [review]
LGTM, just cause we disabled the local tests, can you make sure to do a manual test of the local functionality to triple check all is fine? thanks!

File: internal/cli/deployments/search/indexes/create.go [hunk 1]
```diff
@@ -160,13 +161,25 @@ func (opts *CreateOpts) status(ctx context.Context) (string, error) {
...
+	}
+	return nil, false, nil
+}
+
+func (opts *CreateOpts) watchAtlas(_ context.Context) (any, bool, error) {
+	index, err := opts.store.SearchIndex(opts.ConfigProjectID(), opts.DeploymentName, *opts.index.IndexID)
+	if err != nil {
+		return nil, false, err
+	}
+	if pointer.GetOrZero(index.Status) == "STEADY" {
```
Comment (gssbzn) [inline internal/cli/deployments/search/indexes/create.go:181]:
```suggestion
	if index.GetStatus() == "STEADY" {
```

**Comment by tibulca** [comment]
> LGTM, just cause we disabled the local tests, can you make sure to do a manual test of the local functionality to triple check all is fine? thanks!

for local it's already checked here: https://spruce.mongodb.com/task/mongodb_atlas_cli_master_e2e_local_deployments_atlas_deployments_local_noauth_e2e_patch_07319bbad7ef2b9d622be191947b63f4efc256fd_660c03d3343b7100072f4269_24_04_02_13_10_46?execution=0&sortBy=STATUS&sortDir=ASC

File: internal/cli/deployments/search/indexes/create.go (hunk 1 above)
Comment (tibulca) [reply internal/cli/deployments/search/indexes/create.go]: done, thanks

**Review by blva — DISMISSED** [review]

**Review by gssbzn — APPROVED** [review]

**Review by blva — APPROVED** [review]

**Pull request merged**